### ISSUE TITLE
Error after upload

### ISSUE THREAD

**Issue by pigobyte** [blueimp/jQuery-File-Upload#3199]
Hello everyone ..
   I downloaded the demo and am trying to use the Basic Plus UI because it has the features I need (preview, file size bed, single image cancel, start uplad for all selected files, single and general progress bar) .. I always though the 'error Exception ... "Access to restricted URI denied" code: "1012" nsresult: "0x805303f4 (NS_ERROR_DOM_BAD_URI)" location: "<unknown>"] for each file after uploading .. WHY? : (
I edit

<form id="fileupload" action="server/php/" method="POST" enctype="multipart/form-data">
in
<form id="fileupload" action="server/php/" method="POST" enctype="multipart/form-data">
because my files are in the folder / Applications / XAMPP / xamppfiles / htdocs / Demo
is that correct?

<!DOCTYPE html>

<html lang="en"><head>
<meta http-equiv="content-type" content="text/html; charset=UTF-8">

<meta charset="utf-8">
<title>jQuery File Upload Demo</title>

<link rel="stylesheet" href="jQuery%20File%20Upload%20Demo_files/bootstrap.css">

<link rel="stylesheet" href="jQuery%20File%20Upload%20Demo_files/style.css">

<link rel="stylesheet" href="jQuery%20File%20Upload%20Demo_files/blueimp-gallery.css">

<link rel="stylesheet" href="jQuery%20File%20Upload%20Demo_files/jquery_002.css">
<link rel="stylesheet" href="jQuery%20File%20Upload%20Demo_files/jquery.css">

<noscript><link rel="stylesheet" href="css/jquery.fileupload-noscript.css"></noscript>
<noscript><link rel="stylesheet" href="css/jquery.fileupload-ui-noscript.css"></noscript>
</head>
<body>
<div class="container">
    
    <form id="fileupload" action="server/php/" method="POST" enctype="multipart/form-data">
        
        <div class="row fileupload-buttonbar">
            <div class="col-lg-7">
                
                <span class="btn btn-success fileinput-button">
                    <i class="glyphicon glyphicon-plus"></i>
                    <span>Add files...</span>
                    <input name="files[]" multiple="" type="file">
                </span>
                <button type="submit" class="btn btn-primary start">
                    <i class="glyphicon glyphicon-upload"></i>
                    <span>Start upload</span>
                </button>
                <button type="reset" class="btn btn-warning cancel">
                    <i class="glyphicon glyphicon-ban-circle"></i>
                    <span>Cancel upload</span>
                </button>
                <button type="button" class="btn btn-danger delete">
                    <i class="glyphicon glyphicon-trash"></i>
                    <span>Delete</span>
                </button>
                <input class="toggle" type="checkbox">
                
                <span class="fileupload-process"></span>
            </div>
            
            <div class="col-lg-5 fileupload-progress fade">
                
                <div class="progress progress-striped active" role="progressbar" aria-valuemin="0" aria-valuemax="100">
                    <div class="progress-bar progress-bar-success" style="width:0%;"></div>
                </div>
                
                <div class="progress-extended">&nbsp;</div>
            </div>
        </div>
        
        <table role="presentation" class="table table-striped"><tbody class="files"></tbody></table>
    </form>

```
<br>
<div class="panel panel-default">

    <div class="panel-body">

    </div>
</div>
```

</div>

<div id="blueimp-gallery" class="blueimp-gallery blueimp-gallery-controls" data-filter=":even">
    <div class="slides"></div>
    <h3 class="title"></h3>
    <a class="prev">‹</a>
    <a class="next">›</a>
    <a class="close">×</a>
    <a class="play-pause"></a>
    <ol class="indicator"></ol>
</div>

<script id="template-upload" type="text/x-tmpl">
{% for (var i=0, file; file=o.files[i]; i++) { %}
    <tr class="template-upload fade">
        <td>
            <span class="preview"></span>
        </td>
        <td>
            <p class="name">{%=file.name%}</p>
            <strong class="error text-danger"></strong>
        </td>
        <td>
            <p class="size">Processing...</p>
            <div class="progress progress-striped active" role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="0"><div class="progress-bar progress-bar-success" style="width:0%;"></div></div>
        </td>
        <td>
            {% if (!i && !o.options.autoUpload) { %}
                <button class="btn btn-primary start" disabled>
                    <i class="glyphicon glyphicon-upload"></i>
                    <span>Start</span>
                </button>
            {% } %}
            {% if (!i) { %}
                <button class="btn btn-warning cancel">
                    <i class="glyphicon glyphicon-ban-circle"></i>
                    <span>Cancel</span>
                </button>
            {% } %}
        </td>
    </tr>
{% } %}
</script>

<script id="template-download" type="text/x-tmpl">
{% for (var i=0, file; file=o.files[i]; i++) { %}
    <tr class="template-download fade">
        <td>
            <span class="preview">
                {% if (file.thumbnailUrl) { %}
                    <a href="{%=file.url%}" title="{%=file.name%}" download="{%=file.name%}" data-gallery><img src="{%=file.thumbnailUrl%}"></a>
                {% } %}
            </span>
        </td>
        <td>
            <p class="name">
                {% if (file.url) { %}
                    <a href="{%=file.url%}" title="{%=file.name%}" download="{%=file.name%}" {%=file.thumbnailUrl?'data-gallery':''%}>{%=file.name%}</a>
                {% } else { %}
                    <span>{%=file.name%}</span>
                {% } %}
            </p>
            {% if (file.error) { %}
                <div><span class="label label-danger">Error</span> {%=file.error%}</div>
            {% } %}
        </td>
        <td>
            <span class="size">{%=o.formatFileSize(file.size)%}</span>
        </td>
        <td>
            {% if (file.deleteUrl) { %}
                <button class="btn btn-danger delete" data-type="{%=file.deleteType%}" data-url="{%=file.deleteUrl%}"{% if (file.deleteWithCredentials) { %} data-xhr-fields='{"withCredentials":true}'{% } %}>
                    <i class="glyphicon glyphicon-trash"></i>
                    <span>Delete</span>
                </button>
                <input type="checkbox" name="delete" value="1" class="toggle">
            {% } else { %}
                <button class="btn btn-warning cancel">
                    <i class="glyphicon glyphicon-ban-circle"></i>
                    <span>Cancel</span>
                </button>
            {% } %}
        </td>
    </tr>
{% } %}
</script>

<script src="jQuery%20File%20Upload%20Demo_files/jquery_008.js"></script>

<script src="jQuery%20File%20Upload%20Demo_files/jquery.js"></script>

<script src="jQuery%20File%20Upload%20Demo_files/tmpl.js"></script>

<script src="jQuery%20File%20Upload%20Demo_files/load-image.js"></script>

<script src="jQuery%20File%20Upload%20Demo_files/canvas-to-blob.js"></script>

<script src="jQuery%20File%20Upload%20Demo_files/bootstrap.js"></script>

<script src="jQuery%20File%20Upload%20Demo_files/jquery_009.js"></script>

<script src="jQuery%20File%20Upload%20Demo_files/jquery_002.js"></script>

<script src="jQuery%20File%20Upload%20Demo_files/jquery_006.js"></script>

<script src="jQuery%20File%20Upload%20Demo_files/jquery_010.js"></script>

<script src="jQuery%20File%20Upload%20Demo_files/jquery_005.js"></script>

<script src="jQuery%20File%20Upload%20Demo_files/jquery_011.js"></script>

<script src="jQuery%20File%20Upload%20Demo_files/jquery_004.js"></script>

<script src="jQuery%20File%20Upload%20Demo_files/jquery_007.js"></script>

<script src="jQuery%20File%20Upload%20Demo_files/jquery_003.js"></script>

<script src="jQuery%20File%20Upload%20Demo_files/main.js"></script>

</body></html>

Alternatively I used your plugin Basic (minimal setup guide) .. it works but I do not know how to add the missing functions (preview, file size bed, single image cancel, start uplad for all selected files, single and general progress bar)
<!DOCTYPE HTML>
<html>
<head>
<meta charset="utf-8">
<title>jQuery File Upload Example</title>

<style id="css">
    .bar {
    height: 18px;
    background: green;
}
</style>

</head>
<body>
<input id="fileupload" type="file" name="files[]" data-url="server/php/" multiple>

<script src="//ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>

<script src="js/vendor/jquery.ui.widget.js"></script>

<script src="js/jquery.iframe-transport.js"></script>

<script src="js/jquery.fileupload.js"></script>

<script>
$(function () {
    $('#fileupload').fileupload({
        dataType: 'json',
        done: function (e, data) {
            $.each(data.result.files, function (index, file) {
                $('<p/>').text(file.name).appendTo(document.body);
            });
        }
    });
});

$('#fileupload').fileupload({
    /* ... */
    progressall: function (e, data) {
        var progress = parseInt(data.loaded / data.total * 100, 10);
        $('#progress .bar').css(
            'width',
            progress + '%'
        );
    }
});

$(function () {
    $('#fileupload').fileupload({
        dataType: 'json',
        add: function (e, data) {
            data.context = $('<p/>').text('Uploading...').appendTo(document.body);
            data.submit();
        },
        done: function (e, data) {
            data.context.text('Upload finished.');
        }
    });
});

$(function () {
    $('#fileupload').fileupload({
        dataType: 'json',
        add: function (e, data) {
            data.context = $('<button/>').text('Upload')
                .appendTo(document.body)
                .click(function () {
                    data.context = $('<p/>').text('Uploading...').replaceAll($(this));
                    data.submit();
                });
        },
        done: function (e, data) {
            data.context.text('Upload finished.');
        }
    });
});

</script>

<div id="progress">
    <div class="bar" style="width: 0%;"></div>
</div>

</body> 
</html>

can someone help me?
thanks to all ..

luca
